packages/flutter/lib/src/material/data_table.dart

```diff
@@ -1122,10 +1139,14 @@ class DataTable extends StatelessWidget {
         start: paddingStart,
         end: paddingEnd,
       );
-      if (dataColumnIndex == _onlyTextColumn) {
-        tableColumns[displayColumnIndex] = const IntrinsicColumnWidth(flex: 1.0);
+      if (column.columnWidth != null) {
+        tableColumns[displayColumnIndex] = column.columnWidth!;
       } else {
-        tableColumns[displayColumnIndex] = const IntrinsicColumnWidth();
+        if (dataColumnIndex == _onlyTextColumn) {
+          tableColumns[displayColumnIndex] = const IntrinsicColumnWidth(flex: 1.0);
+        } else {
+          tableColumns[displayColumnIndex] = const IntrinsicColumnWidth();
+        }
```

Comment:
Could we format this as

```dart
if (column.columnWidth != null) {
  // ...
else if (dataColumnIndex == _onlyTextColumn) {
  // ...
} else {
  // ...
}
```

Reply:
Yes, we could format like that. The reason I use the current format is to keep the original logic as much as possible. So, the new logic is something like “if the column width is not null, we will use it, otherwise we will use the default behavior”. Do you think we should update it per your suggestion?

Reply:
Yeah, I think it'd make it more readable, since the goal of each statement body is to assign a value for `tableColumns[displayColumnIndex]`.

Reply:
Done. Thanks for your suggestion!